include/seqan3/range/views/pseudo_random_access.hpp

```diff
@@ -0,0 +1,199 @@
+ * iterator tag.
+ *
+ * ### View properties
+ *
+ * | range concepts and reference_t   | `urng_t` (underlying range type)  | `rrng_t` (returned range type)             |
+ * |----------------------------------|:---------------------------------:|:------------------------------------------:|
+ * | std::ranges::input_range         | *required*                        | *guaranteed*                               |
+ * | std::ranges::forward_range       |                                   | *guaranteed*                               |
+ * | std::ranges::bidirectional_range |                                   | *guaranteed*                               |
+ * | std::ranges::random_access_range |                                   | *guaranteed*                               |
+ * | std::ranges::contiguous_range    |                                   | *preserved*                                |
+ * |                                  |                                   |                                            |
+ * | std::ranges::viewable_range      | *required*                        | *guaranteed*                               |
+ * | std::ranges::view                |                                   | *guaranteed*                               |
+ * | std::ranges::sized_range         |                                   | *preserved*                                |
+ * | std::ranges::common_range        |                                   | *preserved*                                |
+ * | std::ranges::output_range        |                                   | *preserved*                                |
+ * | seqan3::const_iterable_range     |                                   | *preserved*                                |
+ * |                                  |                                   |                                            |
+ * | seqan3::reference_t              |                                   | seqan3::reference_t<urng_t>                |
+ *
+ * See the \link views views submodule documentation \endlink for detailed descriptions of the view properties.
+ *
+ * This adaptor requires that the underlying range models either std::ranges::random_access_range or is one of the
+ * following range types:
+ *
+ *  * seqan3::gap_decorator.
+ *
+ * ### Return type
+ *
+ * | `urng_t` (underlying range type)       | `rrng_t` (returned range type)                       |
+ * |:--------------------------------------:|:----------------------------------------------------:|
+ * | `std::ranges::random_access_range`     | `std::ranges::ref_view<urng_t>`                      |
+ * | `seqan3::gap_decorator`                | `std::ranges::subrange<ra_iterator_t, ra_iterator_t>`|
+ *
+ * The adaptor returns exactly the type specified above. In the second case `ra_iterator_t` refers to
+ * seqan3::gap_decorator::gap_decorator_iterator_ra.
```

Comment:
should we link this if this is hidden from the documentation?

Reply:
it is not hidden in the documentation. I was not sure whether or not this is a bug of doxygen, but albeit that the nested iterator classes are declared in a private section they are both visible in the user documentation. So in order to be clear about the type I decided to then also link to it.

Reply:
Can't you wrap this again into `//!\privatesection`?
Otherwise you could move this documentation into the class and wrap this code into `//!\cond DEV`

Reply:
But honestly, I think it is a good thing to have this iterator public documented. Now most of our concerns are gone. We do not expose an additional begin/end interface but the user can see exactly the type he is getting when using it with the view. It is also documented in the iterator itself of how to get it.

Reply:
fine, I don't care much.